package.json

```diff
@@ -32,20 +33,29 @@
     "@angular-devkit/build-angular": "~0.901.8",
     "@angular/cli": "~9.1.8",
     "@angular/compiler-cli": "~9.1.11",
-    "@types/node": "^12.11.1",
     "@types/jasmine": "~3.5.0",
     "@types/jasminewd2": "~2.0.3",
+    "@types/node": "^12.11.1",
     "codelyzer": "^5.1.2",
+    "husky": "^4.2.5",
     "jasmine-core": "~3.5.0",
     "jasmine-spec-reporter": "~4.2.1",
     "karma": "~5.0.0",
     "karma-chrome-launcher": "~3.1.0",
     "karma-coverage-istanbul-reporter": "~2.1.0",
     "karma-jasmine": "~3.0.1",
     "karma-jasmine-html-reporter": "^1.4.2",
+    "prettier": "^2.0.5",
+    "pretty-quick": "^2.0.1",
     "protractor": "~7.0.0",
     "ts-node": "~8.3.0",
     "tslint": "~6.1.0",
     "typescript": "~3.8.3"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "pretty-quick --staged && ng lint",
+      "pre-push": "npm run test:ci"
```

Comment:
https://stackoverflow.com/questions/56228439/what-is-the-difference-between-git-hooks-and-azure-devops-pipelines

Mye mer robust om det omskrives til en github action

Reply:
Skal se på det

Reply:
`#package.json`
`"test:ci": "ng test --watch=false --browsers=ChromeHeadless",`

Trond har laget en action på testing.

hooksene jeg la til kjøres lokalt for commiting og pushing.

Reply:
Se om du kan gjøre dette i din globale .git config. Så slipper vi legge de til i package.json.

Reply:
Åja, github actions kan gjøre det samme ja... 

https://github.com/marketplace/actions/pre-commit
https://github.com/marketplace/actions/gs-commit-message-checker